_studio/mfx_lib/decode/vp8/include/mfx_vp8_dec_decode_hw.h

```diff
@@ -166,7 +166,7 @@ class MFX_VP8_BoolDecoder
     }
 };
 
-static mfxStatus MFX_CDECL VP8DECODERoutine(void *p_state, void *pp_param, mfxU32 thread_number, mfxU32);
+static inline mfxStatus MFX_CDECL VP8DECODERoutine(void *p_state, void *pp_param, mfxU32 thread_number, mfxU32);
```

Comment:
Fix warning "Wunused-function". Since "inline" not a strict order to compiller and this function called only by address, compiler will not make it inline.

Reply:
no. this is incorrect way to fix this. Actually you need to do the following:
1. Completely remove this declaration and friend declaration from line 217 below
2. Introduce an VideoDECODEVP8_HW class uAPI method, let's say VideoDECODEVP8_HW::TaskRoutine and rewrite P8DECODERoutine as using this new method. You can refer to mpeg2 decoder code:
```
static mfxStatus __CDECL MPEG2TaskRoutine(void *pState, void *pParam, mfxU32 /*threadNumber*/, mfxU32 /*callNumber*/)
{
    VideoDECODEMPEG2InternalBase *decoder = (VideoDECODEMPEG2InternalBase*)pState;

    MFX_AUTO_LTRACE(MFX_TRACE_LEVEL_API, "VideoDECODEMPEG2::TaskRoutine");
    mfxStatus sts = decoder->TaskRoutine(pParam);
    return sts;
}
```